First TA tests and Vision Insurance distinct elements TA.

alexevic opened the pull request:
Tests for Vision Insurance page. All tests work (at least on my pc :) )

KietasRiestutelis committed:
pirmas comitas

KietasRiestutelis committed:
offer title and description test works

KietasRiestutelis committed:
finded btn and btn text

KietasRiestutelis committed:
quizzard page

KietasRiestutelis committed:
validate back button label and click it

KietasRiestutelis committed:
working on offering page object

alexevic committed:
Healtcare page object was created..

..new vision steps added for vision breadcrumbs. Error with quizzard url validation.

KietasRiestutelis committed:
added offerings.page object

alexevic committed:
Merge branch 'dental_testing1' into vision-first-test

alexevic committed:
Tests for all Vision Insurance page elements

Includes only elements that can be found on Vision Insurance page. Code needs refactoring (some variable names can be improved).

Deimokas commented on src/features/dental.feature:
```diff
@@ -3,4 +3,20 @@ Feature: Test for Dental page
+    When the Dental Insurance product page is opened
+    Then i can see in header Offerings and Healthcare links
+    And find Offerings link is working
+
```
unnecessary empty lines

Deimokas commented on src/page-objects/quizzard.page.ts:
```diff
@@ -0,0 +1,53 @@
+    await expectText(this.quizzardFirstQuestionHeaderText, Data.quizzardFirstQuestionHeaderText);
+  }
+
+  async validateQuizzardBackButton() {
```
This validates that button exsists

Deimokas commented on src/page-objects/vision.page.ts:
```diff
@@ -22,19 +43,135 @@ export default class Vision extends Page {
   }
 
-  async validateVisionImage() {
+  async validateVisionHeaderImage() {
```
this expects that image exists

Deimokas commented on src/page-objects/quizzard.page.ts:
```diff
@@ -0,0 +1,53 @@
+    await expectToExist(this.quizzardBackButton);
+  } 
+
+  async validateQuizzardBackButtonText() {
```
Do we really need a method, all logic can be a one liner in step definition

Deimokas commented on src/step-definitions/dental.steps.ts:
```diff
@@ -14,3 +18,50 @@ Then(/^the main Dental title and description should be correct$/, async () => {
+});
+
+Then(/^validate back button label and click it$/, async () => {
+  await _Quizzard.validateQuizzardBackButtonText();
```
Instead of a method it could be just an assert here:
await expectText(this.quizzardBackButtonText, Data.quizzardButtonText);

Deimokas commented on src/step-definitions/dental.steps.ts:
```diff
@@ -14,3 +18,50 @@ Then(/^the main Dental title and description should be correct$/, async () => {
+  await _Quizzard.clickBackButton();
+});
+
+// Then(/^the Dental Insurance product page is opened$/, async () => {
```
why is commented code here?

Commit:
Adding git ignore so that reports would not be picked up every time

Deimokas commented on src/features/vision.feature:
```diff
@@ -3,4 +3,38 @@ Feature: Test for Vision page
+    And correct icon is displayed next to the Vision title
+    And the title and description of What is Vision Insurance? should be correct
+
+  Scenario: Breadcrumb links testing
```
Scenario could be named better, for example:
Scenario: Check that breadcrumb links are correct and navigate user to correct pages

alexevic replied:
Scenario name changed, and scenario was updated.

Deimokas commented on src/features/vision.feature:
```diff
@@ -3,4 +3,38 @@ Feature: Test for Vision page
+
+  Scenario: Breadcrumb links testing
+    Given the user is on the Vision page
+    And the page path elements have correct text
```
What is page path element? Breadcrumb?

alexevic replied:
Yes. Wrote that step before deciding how to call breadcrumbs.

Deimokas commented on src/features/vision.feature:
```diff
@@ -3,4 +3,38 @@ Feature: Test for Vision page
+    And the page path elements have correct text
+    When the user clicks on the Healthcare link
+    Then Healthcare page should be opened
+
```
unnecessary empty line

Deimokas commented on src/features/vision.feature:
```diff
@@ -3,4 +3,38 @@ Feature: Test for Vision page
+    When the user clicks on the Healthcare link
+    Then Healthcare page should be opened
+
+    Given the user is on the Vision page
```
Should not be a second given in the same scenario

Deimokas commented on src/features/vision.feature:
```diff
@@ -3,4 +3,38 @@ Feature: Test for Vision page
+    Then Healthcare page should be opened
+
+    Given the user is on the Vision page
+    And the page path elements have correct text
```
Duplicate step, is this really necessary?

Deimokas commented on src/features/vision.feature:
```diff
@@ -3,4 +3,38 @@ Feature: Test for Vision page
+    When the user clicks on the Offerings link
+    Then Offerings page should be opened
+
+  Scenario: Test breadcrumb icons
```
could be tested in scenario above, could be not. Let's discuss

Deimokas commented on src/features/vision.feature:
```diff
@@ -3,4 +3,38 @@ Feature: Test for Vision page
+    Then the first breadcrumb arrow icon should exist and have a correct image shown
+    And a second breadcrumb arrow icon should exist and have a correct image shown
+
+  Scenario: Enter quiz from Vision page
```
There are multiple places to start quiz, for which place is this test?

alexevic replied:
This was discussed. We test only one element from Vision page that leads to quiz.

alexevic reviewed (commented):
After reviewal, few changes will be made.  Problem areas will be resolved based on the feedback.

alexevic committed:
Refactored Vision page TA

I have made some changes on the things that were described in the comments. Variable naming structure was created. One liner functions deleted. Some of the feature steps were re-written.

Deimokas commented on src/features/vision.feature:
```diff
@@ -1,6 +1,36 @@
+    Given the user is on the Vision page
+    When the Vision Insurance product page is opened
+    And find your product button for Vision has correct text
+    And find your product button for Vision is working
```
instead of working, could be named something like - opens "enter plan name"

Deimokas commented on src/features/vision.feature:
```diff
@@ -1,6 +1,36 @@
+    And user should click on the Offerings link
+    Then Offerings page should be opened
+
+  Scenario: Check if offering header and main content part on Vision Insurance page, has elements and the elements are correct and are displayed correctly
```
Scenario name could be better

Deimokas commented on src/page-objects/offerings.page.ts:
```diff
@@ -0,0 +1,26 @@
+    await expectURL(urls.offerings.base, true);
+  }
+
+  async validateOfferingsTitle() {
```
Is this method really necessary? Could be just an expect in steps
